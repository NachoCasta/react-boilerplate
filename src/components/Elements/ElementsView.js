import React from "react";
import { Flex, Box, CardContainer, Button, Input } from "components/UI";
import List from "./List";

const Elements = ({ onNew, query, onQueryChange }) => (
    <CardContainer width={1} title="Elements" heading>
        <Flex flexWrap="nowrap">
            <Box p={1} width={1}>
                <Input
                    value={query}
                    onChange={onQueryChange}
                    placeholder="Search..."
                />
            </Box>
            <Box p={1}>
                <Button onClick={onNew} type="primary" icon="plus">
                    New
                </Button>
            </Box>
        </Flex>
        <Flex>
            <Box width={1} p={1}>
                <List />
            </Box>
        </Flex>
    </CardContainer>
);

export default Elements;
