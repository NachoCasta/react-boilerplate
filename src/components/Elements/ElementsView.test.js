import React from "react";
import { shallow } from "enzyme";

import ElementsView from "./ElementsView";

describe("ElementsView", () => {
    let component;
    beforeEach(() => {
        component = shallow(<ElementsView />);
    });

    it("renders", () => {
        shallow(<ElementsView />);
    });

    it("matches snapshot", () => {
        expect(component).toMatchSnapshot();
    });
});
