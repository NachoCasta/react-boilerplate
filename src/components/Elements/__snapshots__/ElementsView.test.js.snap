// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementsView matches snapshot 1`] = `ShallowWrapper {}`;
